### PR TITLE
add volatile mode and close background workers

### PR DESCRIPTION
**Pull request by bobotu**
Fix goroutine leak in TiDB's unit tests.
Provide a `VolatileMode` to reduce the disk usages when use it in TiDB's unit tests.

### CODE REVIEW

File: blob.go [hunk 1]
```diff
@@ -460,12 +461,18 @@ type blobGCHandler struct {
...
+	for {
+		select {
+		case discardInfo := <-h.discardCh:
+			h.handleDiscardInfo(discardInfo)
+			err := h.doGCIfNeeded()
+			if err != nil {
+				log.Error(err)
+			}
+		case <-c.HasBeenClosed():
+			return
```
Comment (bobotu) [inline blob.go:475]: Is it safe to ignore the `discardInfo` pending in `h.discardCh`? @coocood

Comment (coocood) [reply blob.go:475]: It is safe if we SignalAndWait it after compactors.

File: db.go [hunk 2]
```diff
@@ -585,7 +590,15 @@ func (db *DB) Close() (err error) {
...
-	if db.opt.CompactL0WhenClose {
+	if db.closers.blobManager != nil {
+		db.closers.blobManager.SignalAndWait()
+		log.Infof("BlobManager finished")
+	}
+	if db.closers.resourceManager != nil {
+		db.closers.resourceManager.SignalAndWait()
+		log.Infof("ResourceManager finished")
+	}
+	if db.opt.CompactL0WhenClose && !db.volatileMode {
```
Comment (coocood) [inline db.go:601]: We should close blobManager and resource manager after final compaction on L0.

Comment (bobotu) [reply db.go:593]: done~

**Comment by coocood** [comment]
go test failed.

**Comment by coocood** [comment]
LGTM

**Pull request merged**